examples/ex1.cpp

```diff
@@ -134,48 +145,67 @@ int main(int argc, char *argv[])
    b->AddDomainIntegrator(new DomainLFIntegrator(one));
    b->Assemble();
 
-   // 7. Define the solution vector x as a finite element grid function
+   // 7. Set device config parameters from the command line options and switch
+   //    to working on the device.
+   Device::Configure(device);
+   Device::Print();
+   Device::Enable();
+
+   // 8. Define the solution vector x as a finite element grid function
    //    corresponding to fespace. Initialize x with initial guess of zero,
    //    which satisfies the boundary conditions.
    GridFunction x(fespace);
    x = 0.0;
 
-   // 8. Set up the bilinear form a(.,.) on the finite element space
+   // 9. Set up the bilinear form a(.,.) on the finite element space
    //    corresponding to the Laplacian operator -Delta, by adding the Diffusion
    //    domain integrator.
-   BilinearForm *a = new BilinearForm(fespace);
+   AssemblyLevel assembly = (pa) ? AssemblyLevel::PARTIAL : AssemblyLevel::FULL;
+   BilinearForm *a = new BilinearForm(fespace, assembly);
    a->AddDomainIntegrator(new DiffusionIntegrator(one));
 
-   // 9. Assemble the bilinear form and the corresponding linear system,
-   //    applying any necessary transformations such as: eliminating boundary
-   //    conditions, applying conforming constraints for non-conforming AMR,
-   //    static condensation, etc.
+   // 10. Assemble the bilinear form and the corresponding linear system,
+   //     applying any necessary transformations such as: eliminating boundary
+   //     conditions, applying conforming constraints for non-conforming AMR,
+   //     static condensation, etc.
    if (static_cond) { a->EnableStaticCondensation(); }
    a->Assemble();
 
-   SparseMatrix A;
    Vector B, X;
+   Operator *A;
+   if (!pa) { A = new SparseMatrix; }
```

Comment:
I believe it is inconsistent that FormLinearSystem expects an initialized object for A in the full assembly case and an uninitialized pointer in the partial assembly case.

Reply:
This is only for backward compatibility.

Reply:
I looks closer and I see the issue now.  You're right there isn't a straightforward way to fix it without breaking bw compatibility